audio_service_web/lib/js/media_session_web.dart

```diff
@@ -1,39 +1,159 @@
+@JS()
 library media_session_web;
 
+import 'dart:js' as js;
 import 'dart:html' as html;
-
 import 'package:js/js.dart';
 
-@JS('setActionHandler')
-external void setActionHandler(String action, Function(ActionResult) callback);
+/// The interface to the Media Session API which allows a web page
+/// to provide custom behaviors for standard media playback interactions,
+/// and to report metadata that can be sent by the user agent
+/// to the device or operating system for presentation in standardized
+/// user interface elements.
+@JS('navigator.mediaSession')
+class MediaSession {
+  /// Returns an instance of [html.MediaMetadata], which contains rich media
+  /// metadata for display in a platform UI.
+  @JS('metadata')
+  external static html.MediaMetadata metadata;
+
+  /// Indicates whether the current media session is playing.
+  ///
+  /// See [MediaSessionPlaybackState] for possible values.
+  @JS('playbackState')
+  external static String playbackState;
+
+  /// Sets an action handler for a media session action, such as play or pause.
+  static void setActionHandler(
+    String action,
+    MediaSessionActionHandler callback,
+  ) =>
+      _setActionHandler(action, js.allowInterop(callback));
+
+  @JS('setActionHandler')
+  external static void _setActionHandler(
+    String action,
+    MediaSessionActionHandler callback,
+  );
+
+  /// Sets the current playback position and speed of the media currently being presented.
+  @JS('setPositionState')
+  external static void setPositionState(MediaPositionState? state);
+}
+
+/// Media session playback state types.
+///
+/// See [MediaSession.playbackState].
+abstract class MediaSessionPlaybackState {
+  MediaSessionPlaybackState._();
+
+  /// The browsing context doesn't currently know the current playback
+  /// state, or the playback state is not available at this time.
+  static const none = 'none';
+
+  /// The browser's media session is currently paused. Playback may be resumed.
+  static const paused = 'paused';
+
+  /// The browser's media session is currently playing media, which can be paused.
+  static const playing = 'playing';
+}
+
+/// Actions that the user may perform in a media session.
+abstract class MediaActions {
+  MediaActions._();
+
+  static const play = 'play';
+  static const pause = 'pause';
+  static const seekbackward = 'seekbackward';
+  static const seekforward = 'seekforward';
+  static const previoustrack = 'previoustrack';
+  static const nexttrack = 'nexttrack';
+  static const skipad = 'skipad';
+  static const stop = 'stop';
+  static const seekto = 'seekto';
+  static const togglemicrophone = 'togglemicrophone';
+  static const togglecamera = 'togglecamera';
+  static const hangup = 'hangup';
+}
 
-@JS('setPositionState')
-external void setPositionState(PositionState state);
+/// A callback signature for the [MediaSession.setActionHandler].
+typedef MediaSessionActionHandler = Function(MediaSessionActionDetails);
```

Comment:
According to the spec this is details "MUST be provided" https://w3c.github.io/mediasession/#dictdef-mediasessionactiondetails